garden-backend-service/src/api/routes/gardens.py

```diff
@@ -81,6 +81,34 @@ async def search_gardens(
     return result.all()
 
 
+@router.post(
+    "/gardens-using-functions",
```

Comment:
I think this is a little wonky from a REST perspective. This is a retrieval, so it seems like it should be a GET. And /gardens-using-functions aren't distinct resources from /gardens. Can this functionality be added to GET /gardens with a query param?